File: frontend/src/interfaces/OrderItem.ts
import type { Order } from "./Order";
import type { KeyGame } from "./KeyGame";

export interface OrderItem {
  ID: number;
  unit_price: number;
  qty: number;
  line_discount: number;
  line_total: number;
  order_id: number;
  game_key_id?: number | null;
  key_game?: KeyGame;
}

export interface CreateOrderItemRequest {
  unit_price: number;
  qty: number;
  line_discount: number;
  line_total: number;
  order_id: number;
  game_key_id?: number | null;
}

export interface UpdateOrderItemRequest {
  ID: number;
  unit_price?: number;
  qty?: number;
  line_discount?: number;
  line_total?: number;
  order_id?: number;
  game_key_id?: number | null;
}

File: backend/entity/order_item.go
// entity/order_item.go
package entity

import "gorm.io/gorm"

type OrderItem struct {
	gorm.Model

	OrderID uint   `json:"order_id"`
	Order   *Order `gorm:"foreignKey:OrderID" json:"order,omitempty"`

	GameID uint  `json:"game_id"`
	Game   *Game `gorm:"foreignKey:GameID" json:"game,omitempty"`

	QTY          int     `json:"qty"`
	UnitPrice    float64 `json:"unit_price"`
	LineDiscount float64 `json:"line_discount"`
	LineTotal    float64 `json:"line_total"`
}
